libs/stats/odc/stats/_fc_percentiles.py

```diff
@@ -89,9 +89,23 @@ def input_data(self, task: Task) -> xr.Dataset:
 
         return xx
 
-    def reduce(self, xx: xr.Dataset) -> xr.Dataset:        
+    @staticmethod
+    def reduce(xx: xr.Dataset) -> xr.Dataset: 
+        # not 255 & True => 1
+        # not 255 & False => 1
+        # 255 & False => 2  
+        # 255 + True => 3
+
         yy = xr_percentile(xx.drop_vars(["wet"]), [0.1, 0.5, 0.9], nodata=255)
-        yy["is_ever_wet"] = _or_fuser(xx["wet"]).squeeze(xx["wet"].dims[0], drop=True)
+        is_ever_wet = _or_fuser(xx["wet"]).squeeze(xx["wet"].dims[0], drop=True)
+
+        band, *bands = [band for band in yy.data_vars.keys() if band != "wet"]
```

Comment:
`yy` doesn't have `wet` band anyway

Reply:
Probably easier approach is something like this:

```python
    all_invalid = (yy.to_array() == yy.nodata).sum(axis=0, dtype='int8') > 0
```

- turn band into dimension
- turn array into boolean array T if `nodata`
- sum boolean array along band dimension
- if result more than 0 there is at least one band with nodata value

Reply:
it probably doesn't make much difference either way but the current thing doesn't do any type casting - so I'll leave as is for now